### PR TITLE
Update Windows setup and build README.md

### LINKED ISSUE #4: Outdated Windows build instructions strawberry-msvc/README.md
**Describe the bug**
The Windows build instruction is outdated and broken:

The [README.md](https://github.com/strawberrymusicplayer/strawberry-msvc/blob/master/README.md) refers to `copy-deps-msvc-x64-debug-qt6.bat` and `copy-deps-msvc-x64-release-qt6.bat` which are removed in this commit [Remove copy deps files](https://github.com/strawberrymusicplayer/strawberry-msvc/commit/47dd966512acacb1e7d10433a7202b8332d12dff).

However, these build are required before starting the NSIS setup build. Additionally, several dependencies are missing.

**To Reproduce**
Follow instructions `README.md`.

**Expected behavior**
Clear build description for Windows, no errors.

**Screenshots:**
N.A.

**System Information:**
 - Operating system: Windows
 - Strawberry Version: github.com/strawberrymusicplayer/strawberry-msvc 367396afc61f9731b34c98a935d7b98be70eaa6f

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by Erriez**
- Refactor Windows setup and build instructions
- Add NSIS installer to downloads

Closes #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,92 +1,150 @@
...
-This guide uses Visual Studio 2022 to build Strawberry as well all required libraries.
+This guide uses Visual Studio 2022 to build Strawberry Music Player as well as required dependencies.
 
 These instructions are provided as-is, they are primarily intended for developers working on Strawberry on Windows.
 
 We do not offer support to users for building Strawberry on Windows.
 
 Build tools (Git, CMake, Meson, Perl, Python, etc) versions are not regulary updated, so make sure you bump the versions in versions.bat before you run install.bat.
 
-### Requirements
```
Comment (jonaski) [inline README.md:12]: I'd like to keep the requirements list.

File: README.md [hunk 2]
```diff
@@ -1,92 +1,150 @@
...
+- Start | Visual Studio 2022
+- Open a local folder `C:\data\projects\strawberry\msvc_\downloads\strawberry`
+- Toolbar Project | CMake Settings:
+    - Create new `x64-Release`
+    - Select Configurations `x64-Release`
+    - Add `CMake command arguments`: `-DARCH=x86_64 -DICU_ROOT=c:\\strawberry_msvc_x86_64_release`
+    - Save the changes
+    - In the toolbar select `x64-Release`
+    - Project | Delete Cache and Reconfigure
+    - Uncheck: `ENABLE_WIN32_CONSOLE`
```
Comment (jonaski) [inline README.md:147]: Add `-DENABLE_WIN32_CONSOLE=OFF` to CMake arguments instead of this.

File: README.md [hunk 3]
```diff
@@ -1,92 +1,150 @@
...
+- Toolbar: Build | Build All
+
+Set breakpoint and press F5 to step through the code.
```
Comment (jonaski) [inline README.md:150]: This is wrong, F5 is continue, F10 is step.

File: README.md [hunk 4]
```diff
@@ -1,92 +1,150 @@
...
-
-For debug:
-
-    -DARCH=x86_64 -DICU_ROOT=c:\\strawberry_msvc_x86_64_debug
-
-
-For release:
-
-    -DARCH=x86_64 -DICU_ROOT=c:\\strawberry_msvc_x86_64_release
+## Links
```
Comment (jonaski) [inline README.md:11]: We don't need these links

File: README.md [hunk 5]
```diff
@@ -1,92 +1,150 @@
...
+        - In first dialog:
+            - Check: `Add python.exe to PATH`
+            - Click `Install Now` (to use defaults)
+        - In dialog `Setup was successful`:
+            - Click: `Disable path length limit`
+    - Install `strawberry-perl-<VERSION>-64bit.msi`
+    - Extract `win_flex_bison-<VERSION>.zip`
+- **Recommended: When building dependencies from source**:
+    - Install `nasm-<VERSION>-installer-x64.exe`
+        - Run as normal user and click OK
```
Comment (jonaski) [inline README.md:54]: I suggest to change this to a normal installation using administrator to ``C:\Program Files\NASM``

File: README.md [hunk 6]
```diff
@@ -1,92 +1,150 @@
...
+
+Delete `pkg-config` files to prevent conflicts with Strawberry's own `pkg-config`:
+- Delete `C:\strawberry\perl\bin\pkg-config`
+- Delete `C:\strawberry\perl\bin\pkg-config.bat`
+
+Windows Settings | System | About | Advanced system settings | Tab Advanced | Environment Variables:
+- System variables | Path | Edit:
+    - Delete: `C:\Strawberry\c\bin` (To prevent conflicts with other utilities)
+    - Add: `C:\Program Files\Git\bin` (This is for `sed` and `patch` utilities)
+    - Add: `C:\Users\user\AppData\Local\bin\NASM`
```
Comment (jonaski) [inline README.md:75]: `C:\Program Files\NASM`

File: README.md [hunk 7]
```diff
@@ -1,92 +1,150 @@
...
+    - Add: `C:\data\projects\strawberry\msvc_\downloads\win_flex_bison-<VERSION>`
```
Comment (jonaski) [inline README.md:76]: Anything under the msvc_ directory is used by the bat files and shouldn't be used for permanent installation, I suggest to change this to `c:\win_flex_bison`, it's already done by install.bat

File: README.md [hunk 8]
```diff
@@ -1,92 +1,150 @@
...
+Strawberry and Windows installer executables are generated in:
+- For debug: `C:\data\projects\strawberry\msvc_\build_debug\strawberry\build\`
+- For release: `C:\data\projects\strawberry\msvc_\build_release\strawberry\build\`
+
+## Debug Strawberry with Visual Studio
+
+Debug build:
+- Make sure path `C:\strawberry_msvc_x86_64_debug\bin` is set in environment variables
+- Start | Visual Studio 2022
+- Open a local folder `C:\data\projects\strawberry\msvc_\downloads\strawberry`
```
Comment (jonaski) [inline README.md:129]: Don't use this directory as it should be seen as a temporary directory used by the bat files, use: ``C:\data\projects\strawberry\strawberry``.

File: README.md [hunk 9]
```diff
@@ -1,92 +1,150 @@
...
+- [Github MSVC](https://github.com/strawberrymusicplayer/strawberry-msvc)
+- [Github MSVC prebuilt binary dependencies](https://github.com/strawberrymusicplayer/strawberry-msvc-dependencies)
+
+## Manual installation on Windows
+
+Download and install manually:
+- [Git](https://git-scm.com/downloads)
+    - Default settings can be used
+- [Visual Studio 2022 Community](https://visualstudio.microsoft.com/vs/)
+    - Select `Desktop development with C++`
```
Comment (jonaski) [inline README.md:26]: Is this enough, does it install C++ CMake tools for Windows?

What about Qt VS Tools for Visual Studio 2022?

File: README.md [hunk 10]
```diff
@@ -1,92 +1,150 @@
...
+
+## Windows setup
+
+Open a Windows Terminal and type:
```
Comment (jonaski) [inline README.md:30]: Command prompt

File: README.md [hunk 11]
```diff
@@ -1,92 +1,150 @@
...
+
+Prebuilt MSVC binaries can be optionally used to speed-up the build process. When this step is skipped, all libraries and dependencies are build from source.
+
+To use prebuilds, download the following `tar.xz` files from Github [strawberry-msvc-dependencies/releases](https://github.com/strawberrymusicplayer/strawberry-msvc-dependencies/releases):
+- For debug, extract `strawberry-msvc-x86_64-debug.tar.xz` to `C:\strawberry_msvc_x86_64_debug`
+- For release, extract `strawberry-msvc-x86_64-release.tar.xz` to `C:\strawberry_msvc_x86_64_release`
+
+## Build Strawberry from source including dependencies
+
+- Set one debug or release in `Path` environment variable:
```
Comment (jonaski) [inline README.md:102]: `PATH `not `Path`

File: README.md [hunk 12]
```diff
@@ -1,92 +1,150 @@
...
+```
+
+Strawberry and Windows installer executables are generated in:
+- For debug: `C:\data\projects\strawberry\msvc_\build_debug\strawberry\build\`
+- For release: `C:\data\projects\strawberry\msvc_\build_release\strawberry\build\`
+
+## Debug Strawberry with Visual Studio
+
+Debug build:
+- Make sure path `C:\strawberry_msvc_x86_64_debug\bin` is set in environment variables
```
Comment (jonaski) [inline README.md:127]:
```suggestion
- Make sure the path `C:\strawberry_msvc_x86_64_debug\bin` is added to the `PATH` environment variable before starting Visual Studio 2022.
```

File: README.md [hunk 13]
```diff
@@ -1,92 +1,150 @@
...
+## Links
+
+- [Strawberry Music Player Website](https://www.strawberrymusicplayer.org)
+- [Wiki](https://wiki.strawberrymusicplayer.org/wiki/Main_Page)
+- [Forum](https://forum.strawberrymusicplayer.org)
+- [Github sources](https://github.com/strawberrymusicplayer/strawberry)
+- [Github MSVC](https://github.com/strawberrymusicplayer/strawberry-msvc)
+- [Github MSVC prebuilt binary dependencies](https://github.com/strawberrymusicplayer/strawberry-msvc-dependencies)
+
+## Manual installation on Windows
```
Comment (jonaski) [inline README.md:20]: I think this should be under Windows setup too. Also, probably the PATH settings under it's own subtitle, perhaps:
- Installing requirements
- Preparing system for building

File: README.md [hunk 14]
```diff
@@ -1,92 +1,150 @@
...
+For release build:
+```
+.\build.bat release
+```
+
+Strawberry and Windows installer executables are generated in:
+- For debug: `C:\data\projects\strawberry\msvc_\build_debug\strawberry\build\`
+- For release: `C:\data\projects\strawberry\msvc_\build_release\strawberry\build\`
+
+## Debug Strawberry with Visual Studio
```
Comment (jonaski) [inline README.md:124]: Setting up Strawberry for development in Visual Studio 2022

(Not just debugging)

File: README.md [hunk 15]
```diff
@@ -1,92 +1,150 @@
...
+    - Install `7z<VERSION>-x64.exe`
+    - Install `cmake-<VERSION>-windows-x86_64.msi`
+    - Install `python-<VERSION>-amd64.exe`
+        - In first dialog:
+            - Check: `Add python.exe to PATH`
+            - Click `Install Now` (to use defaults)
+        - In dialog `Setup was successful`:
+            - Click: `Disable path length limit`
+    - Install `strawberry-perl-<VERSION>-64bit.msi`
+    - Extract `win_flex_bison-<VERSION>.zip`
```
Comment (jonaski) [inline README.md:51]: Extract to `c:\win_flex_bison`

**Review by jonaski — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Erriez) [reply README.md:12]: Restored in 5d749b413e845a23772b20f69a6b5b8aa638ae6f.

File: README.md (hunk 11 above)
Comment (Erriez) [reply README.md:119]: On Windows 11 this changed to `Path`:

image

Both `set Path` and `set PATH` works as it is not case sensitive like Linux.

File: README.md (hunk 4 above)
Comment (Erriez) [reply README.md]: Removed in c913f626653c89f59e36a5162179a8709be0fa6e.

File: README.md (hunk 3 above)
Comment (Erriez) [reply README.md]: Reworded in 34af329fe7723cf1eda80f8c60be094db344f154.

File: README.md (hunk 12 above)
Comment (Erriez) [reply README.md]: Fixed in 020ff659e85a345cc9c52eaa008c3ef6938cf9e0.

File: README.md (hunk 14 above)
Comment (Erriez) [reply README.md]: Changed in 999b2013e9fc94c493ce2420c16f45a53608195a.

File: README.md (hunk 15 above)
Comment (Erriez) [reply README.md]: Good point: I've tested by setting an environment variable to `C:\data\projects\strawberry\msvc_\downloads\win_flex_bison-<VERSION>` and is no longer needed by extracting to `C:\win_flex_bison`. :+1:

Fixed in 2c2b21d20ebdc7232f6079114a627052fd7f1cb3.

File: README.md (hunk 8 above)
Comment (Erriez) [reply README.md:136]: I assume your idea is to add a clone of the Strawberry source repository in section `Windows setup`? I don't have it on my disk after running `download.bat` and `build.bat`:

```diff
mkdir C:\data\projects\strawberry
cd C:\data\projects\strawberry
+git clone git@github.com:strawberrymusicplayer/strawberry.git
git clone https://github.com/strawberrymusicplayer/strawberry-msvc.git
```

File: README.md (hunk 7 above)
Comment (Erriez) [reply README.md]: Path `c:\win_flex_bison` is used by `build.bat`, so removed environment variable.

Note: The `install.bat` does not install Python, Meson, NASM and NSIS correctly and works only with a manual installation, so did not mention this script. Maybe for another MR to fix.

Path environment variable to `win_flex_bison` removed in 2c2b21d20ebdc7232f6079114a627052fd7f1cb3.

File: README.md (hunk 2 above)
Comment (Erriez) [reply README.md]: Yes, your suggestion is better.

Changed in 1bfe2ab0a2f5d10a63b05d64d50d919a2a6cfee8.

File: README.md (hunk 5 above)
Comment (Erriez) [reply README.md]: Correct, path `C:\Program Files\NASM` is used in `build.bat`, so there is no environment variable needed.

Fixed in 83adf8f6c50e2a7f070a760bc17b2987531e0229.

File: README.md (hunk 6 above)
Comment (Erriez) [reply README.md]: Environment variable to `NASM` removed, but I have to test it.

Fixed in 83adf8f6c50e2a7f070a760bc17b2987531e0229.

File: README.md (hunk 9 above)
Comment (Erriez) [reply README.md:34]: > Is this enough, does it install C++ CMake tools for Windows?

Yes, when checking checkbox `Desktop devleopment with C++`, it installed all features/dependencies listed in the screenshot below, including `C++ CMake tools for Windows`. No other checkboxes are needed:

image

> What about Qt VS Tools for Visual Studio 2022?

No, this is not installed automatically and not needed to build Strawberry debug or release. I think it is suseful for development.

Installation is described here: https://www.qt.io/blog/qt-vs-tools-for-visual-studio-2022.

Microsoft likes it to change interfaces, so in my version it looks like this:

image

File: README.md (hunk 13 above)
Comment (Erriez) [reply README.md]: Subtitles restructured in 93eda230a80a2b37133c5e9e79d7c008b80c9b46.

File: README.md (hunk 10 above)
Comment (jonaski) [reply README.md]: I still have Windows 10 :)

**Review by jonaski — APPROVED** [review]
Thanks, I think this is good now.
Please squash all the commits and I'll merge this.

**Comment by Erriez** [comment]
Squashed `README.md` commits, ready to merge.

Thanks for your great help! 👍

**Pull request merged**